src/libraries/System.Private.Xml/src/System/Xml/XPath/Internal/FollowingQuery.cs

```diff
@@ -36,7 +38,7 @@ public override XPathNavigator Advance()
                 XPathNavigator prev;
                 do
                 {
-                    prev = _input.Clone();
+                    prev = _input!.Clone();
```

Comment:
this shouldn't be required - perhaps IsDescendant should have some annotation like `[NotNullIf(true)]`

Reply:
Nice.